pkg/kubectl/cmd/convert_test.go

```diff
@@ -0,0 +1,57 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"bytes"
+	"strings"
+	"testing"
+
+	"k8s.io/client-go/rest/fake"
+	cmdtesting "k8s.io/kubernetes/pkg/kubectl/cmd/testing"
+)
+
+func TestConvertObject(t *testing.T) {
+	f, tf, _, _ := cmdtesting.NewAPIFactory()
+	tf.Printer = &testPrinter{}
+	//deleted := false
+	tf.UnstructuredClient = &fake.RESTClient{}
+	tf.Namespace = "test"
+	buf := bytes.NewBuffer([]byte{})
+
+	cmd := NewCmdConvert(f, buf)
+	cmd.Flags().Set("filename", "../../../test/fixtures/pkg/kubectl/cmd/convert/appsdeployment.yaml")
+	cmd.Flags().Set("output-version", "extensions/v1beta1")
+	cmd.Flags().Set("local", "true")
+	cmd.Run(cmd, []string{})
+
+	if !strings.Contains(buf.String(), "extensions/v1beta1") {
```

Comment:
Not just the kind/version… need to make sure the conversion was actually correct. Check a field that depends on an internal conversion to be correct (can’t just be done by reflection between field names and types)

A good example is a v1 HPA object with targetCPUUtilizationPercentage converting to the corresponding v2beta1 metric spec source

Reply:
I will add another one to check field are correct.
I want to keep this one to make sure cross-group conversion works.